### PR TITLE
DO NOT MERGE SC-12 Use SSM scheduler for maintenance tasks

### PR DESCRIPTION
**Pull request by ahayden**
Moved repos: https://github.com/Sage-Bionetworks/aws-infra/pull/233

### CODE REVIEW

File: templates/maintenance.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    Properties:
+      Name: DefaultPatchGroupTarget
+      Description: "Instances tagged PatchGroup=prod-default"
+      ResourceType: INSTANCE
+      WindowId:
+        Ref: DefaultMaintenanceWindow
+      Targets:
+        - Key: 'tag:PatchGroup'
+          Values:
+            - 'prod-default'  #Current we have only one group
```
Comment (zaro0508) [inline templates/maintenance.yaml:25]: hmm, i'm wondering if there's a better tag name for this?  we may want to apply this maintenance to instances in a test account in which case the `prod` portion may not apply anymore?  how about something like `apply-patches-biweekly`, or `patch-me-biweekly`?

File: templates/maintenance.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      Cutoff: 1
+      Schedule: 'cron(00 10 ? * SUN/2 *)' #1000UTC Every other Sunday
+      ScheduleTimezone: 'UTC'
+      AllowUnassociatedTargets: false
+  DefaultPatchGroupTarget:
+    Type: 'AWS::SSM::MaintenanceWindowTarget'
+    Properties:
+      Name: DefaultPatchGroupTarget
+      Description: "Instances tagged PatchGroup=prod-default"
+      ResourceType: INSTANCE
```
[the remark this answers is not in the document]
Comment (zaro0508) [reply templates/maintenance.yaml:19]: I misunderstood `RESOURCE_GROUP` in this context.  it's not a group of instances but rather a group of hetrogeneous resources. https://docs.aws.amazon.com/systems-manager/latest/userguide/mw-cli-tutorial-targets-examples.html

For this it seems like `INSTANCE` makes more sense.

File: templates/maintenance.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
+Description: Maintenance tasks for scipoolprod instances
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  DefaultMaintenanceWindow:
+    Type: 'AWS::SSM::MaintenanceWindow'
+    Properties:
+      Name: DefaultMaintenanceWindow
```
[the remark this answers is not in the document]
Comment (ahayden) [reply templates/maintenance.yaml:7]: Agreed!

File: templates/maintenance.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+                - ssm.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+  MaintenanceWindowAgentUpdateTask:
+    Type: 'AWS::SSM::MaintenanceWindowTask'
+    Properties:
```
[the remark this answers is not in the document]
Comment (ahayden) [reply templates/maintenance.yaml:56]: The agent update task doesn't take any parameters

File: templates/maintenance.yaml (hunk 1 above)
Comment (ahayden) [reply templates/maintenance.yaml:25]: I'd like the abstract the schedule from the tags, because we might change the schedule at some point.

**Pull request closed without merging**